### PR TITLE
Add new lint rules documentation

### PR DESCRIPTION
**Pull request by jianyexi**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: documentation/openapi-authoring-automated-guidelines.md [hunk 1]
```diff
@@ -2443,19 +2448,24 @@ The following would be invalid:
 ```
 Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
 
-### <a name="r4009" ></a>R4009 RequiredSystemDataInNewApiVersions
+### <a name="r4009" ></a>R4009 RequiredSystemData
```
Comment (josefree) [inline documentation/openapi-authoring-automated-guidelines.md:2451]: suggest name: RequiredReadOnlySystemData

File: documentation/openapi-authoring-automated-guidelines.md [hunk 2]
```diff
@@ -2443,19 +2448,24 @@ The following would be invalid:
...
+The property systemData in the response of operation:'${operationId}' is not read only. Please add the readonly for the systemData.
+
+**Description** : Per [common-api-contracts](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md#system-metadata-for-all-azure-resources), all Azure resources should implement the `systemData` object property in new api-version. The systemData should be readonly.
 
 **CreatedAt**: May 21, 2020
 
-**LastModifiedAt**: May 21, 2020
+**LastModifiedAt**: February 26, 2021
 
 **How to fix the violation**: For each response in the GET/PUT/PATCH operation add the systemData object.
```
Comment (josefree) [inline documentation/openapi-authoring-automated-guidelines.md:2470]: For each response in the GET/PUT/PATCH operation,  add the systemData object. 

the sentence is confusing,  1. if systemData must be ReadOnly ? 2. if it must be a property of Object type?

File: documentation/openapi-authoring-automated-guidelines.md [hunk 3]
```diff
@@ -3260,3 +3271,192 @@ resourceDefinition": {
...
+
+### <a name="r4032"></a>R4032 MissingXmsErrorResponse
+
+**Category** : SDK Error
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** : The response code {0} is defined without a x-ms-error-response.
+
+**Description** :  If defines response code 4xx or 5xx ,  x-ms-error-response:true is required. There is one exception: a HEAD operation with 404 SHOULD have x-ms-error-response:false.
```
Comment (josefree) [inline documentation/openapi-authoring-automated-guidelines.md:3283]: why 404 SHOULD have exception, could you add reason in WHY section to raise attention?

File: documentation/openapi-authoring-automated-guidelines.md [hunk 4]
```diff
@@ -3260,3 +3271,192 @@ resourceDefinition": {
...
+
+**CreatedAt**: February 23, 2021
+
+**LastModifiedAt**: February 23, 2021
+
+**Why this rule is important**: As some SDK may treat the 4xx or 5xx as exceptional code, if don't specified x-ms-error-response:true, the SDK will not handle the error schema correctly instead it will throw an exception.
```
Comment (josefree) [inline documentation/openapi-authoring-automated-guidelines.md:3289]: need to clarify why 404 should be an exception

File: documentation/openapi-authoring-automated-guidelines.md [hunk 5]
```diff
@@ -3260,3 +3271,192 @@ resourceDefinition": {
...
+
+The following would be valid:
+
+```json
+ "responses": {
+    "404": {
+      "description": "Not Found",
+      "x-ms-error-response": true
+    }
+ }
```
Comment (josefree) [inline documentation/openapi-authoring-automated-guidelines.md:3301]: seems conflict with the description:

There is one exception: a HEAD operation with 404 SHOULD have x-ms-error-response:false.

File: documentation/openapi-authoring-automated-guidelines.md [hunk 6]
```diff
@@ -3260,3 +3271,192 @@ resourceDefinition": {
...
+
+### <a name="r4033"></a>R4033 UniqueModelName
+
+**Category** : SDK Error
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** :  The model name {0} is duplicated with {1} .
+
+**Description** :  Do not use case sensitivity to differentiate model names.
```
Comment (josefree) [inline documentation/openapi-authoring-automated-guidelines.md:3323]: Do not rely on case sensitivity to differentiate different **model**.

I am not sure reader has common sense on the term 'model', in other section of the document, I see 'object'.

File: documentation/openapi-authoring-automated-guidelines.md [hunk 7]
```diff
@@ -3260,3 +3271,192 @@ resourceDefinition": {
...
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="r4034"></a>R4034 AzureResourceTagsSchemaValidation
+
+**Category** : SDK Error
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** :  The resource "{0}" property tags schema is different with the common type definition.
```
Comment (josefree) [inline documentation/openapi-authoring-automated-guidelines.md:3371]: The property tags in the resource "{0}" does not follow/adopt the common type definition.

File: documentation/openapi-authoring-automated-guidelines.md [hunk 8]
```diff
@@ -3260,3 +3271,192 @@ resourceDefinition": {
...
+
+**Description** :  This rule is to check if the tags definition of a resource is same with the common tags definition.
+
+**CreatedAt**: February 23, 2021
+
+**LastModifiedAt**: February 23, 2021
+
+**Why this rule is important**: It will block the SDK generation for terraform, as it's only accepted that the Golang type for tags is map[string]*string .
```
Comment (josefree) [inline documentation/openapi-authoring-automated-guidelines.md:3379]: Terraform

File: documentation/openapi-authoring-automated-guidelines.md [hunk 9]
```diff
@@ -3260,3 +3271,192 @@ resourceDefinition": {
...
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="r4035"></a>R4035 PrivateEndpointResourceSchemaValidation
+
+**Category** : SDK Error
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** :  The private endpoint model "{0}" schema is different with the common type definition.
```
Comment (josefree) [inline documentation/openapi-authoring-automated-guidelines.md:3417]: The private endpoint model "{0}" does not follow/adopt the common type definition.

Comment (josefree) [reply documentation/openapi-authoring-automated-guidelines.md:3417]: or 'conform to' as being used below

File: documentation/openapi-authoring-automated-guidelines.md (hunk 6 above)
Comment (jianyexi) [reply documentation/openapi-authoring-automated-guidelines.md]: object is a swagger/json type while model indicates definition

File: documentation/openapi-authoring-automated-guidelines.md (hunk 8 above)
Comment (jianyexi) [reply documentation/openapi-authoring-automated-guidelines.md]: fixed

File: documentation/openapi-authoring-automated-guidelines.md (hunk 3 above)
Comment (jianyexi) [reply documentation/openapi-authoring-automated-guidelines.md]: Sure, I will confirm with the initiator of this rule

File: documentation/openapi-authoring-automated-guidelines.md (hunk 5 above)
Comment (jianyexi) [reply documentation/openapi-authoring-automated-guidelines.md:3301]: fixed

File: documentation/openapi-authoring-automated-guidelines.md (hunk 9 above)
Comment (jianyexi) [reply documentation/openapi-authoring-automated-guidelines.md]: fixed

File: documentation/openapi-authoring-automated-guidelines.md (hunk 2 above)
Comment (jianyexi) [reply documentation/openapi-authoring-automated-guidelines.md]: Yes for the two questions, I updated it

File: documentation/openapi-authoring-automated-guidelines.md (hunk 1 above)
Comment (jianyexi) [reply documentation/openapi-authoring-automated-guidelines.md]: good point

File: documentation/openapi-authoring-automated-guidelines.md (hunk 4 above)
Comment (jianyexi) [reply documentation/openapi-authoring-automated-guidelines.md:3289]: fixed

**Review by josefree — APPROVED** [review]

**Pull request merged**